weldx/time.py

```diff
@@ -490,3 +491,17 @@ def union(times=List[Union[types_time_like, "Time"]]) -> Time:
             (Time(time).as_pandas_index() for time in times),
         )
         return Time(pandas_index)
+
+
+class TimeDependent(ABC):
+    """An abstract base class that describes a common interface of time dep. classes."""
+
+    @property
+    @abstractmethod
+    def time(self) -> Union[Time, None]:
+        """Get the classes time component"""
+
+    @property
+    @abstractmethod
+    def reference_time(self) -> Union[Timestamp, None]:
+        """Returns the reference timestamp if the time data is absolute."""
```

Comment:
Now every class that inherits from `TimeDependent` must implement those two methods (or any other we want to enforce). Otherwise we get the error:

~~~
TypeError: Can't instantiate abstract class SomeClass with abstract method reference_time
~~~

Unfortunately, the specified return type is not enforced. However, as soon as we get mypy up and running, it will complain about using other return types than the one given by the ABC:

~~~
error: Return type "str" of "time" incompatible with return type "Time" in supertype "TimeDependent"
~~~